packages/pwa-kit-react-sdk/src/ssr/universal/extensibility/application-extension.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {IRouteConfig} from './types'
+
+/**
+ * An abstract class representing an Application Extension. This class provides
+ * foundational methods and properties for extending an application with additional
+ * configuration and routing capabilities. It is designed to be subclassed
+ * by other Application Extensions that need to augment the base application, particularly
+ * during server and client-side rendering.
+ *
+ * @abstract
+ */
+export default abstract class ApplicationExtension {
+    private config: Record<string, any>
+
+    /**
+     * Constructs a new instance of the ApplicationExtension class.
+     *
+     * @param {string} [name] - The name of the extension. Defaults to the class name if not provided.
+     * @param {Record<string, any>} [config] - The configuration object used to set up the extension.
+     */
+    constructor(config?: any) {
```

Comment:
There is no `name` parameter used in this constructor.

Reply:
I see that there is jsdoc mixed with typescript code in this file. As a result, there are duplicate types: the `@param` ones in the jsdoc comment and the one in typescript code `(config?: any)`.

The jsdoc comment does not need to define the param types. But we can still use it to mention the purpose "Constructs a new instance of the ApplicationExtension class".

Reply:
Good call out. I had a friend generate those for me. Lets follow googles style guide and not include them as you mentioned, they are redundant. 

```
JSDoc type annotations
JSDoc type annotations are redundant in TypeScript source code. Do not declare types in @param or @return blocks, do not write @implements, @enum, @private, @override etc. on code that uses the implements, enum, private, override etc. keywords.
```